### ISSUE TITLE
[ Website ]: Not all accordions work without JavaScript

### ISSUE THREAD

**Issue by rogeruiz** [18F/web-design-standards#1245]
## Issue type
* Website

## Description

Not all the accordions on the site work without JavaScript, #1224 covers the accordion page, but the Typography page doesn't open. And I'm assuming that there may be other pages where accordion's are collapsed by default.

## Steps to reproduce the issue

1. Turn off JavaScript
1. Navigation to https://wds-microsite-release.apps.cloud.gov/visual-style/#pairings
1. Click the expand button on the accordion
1. Witness the blatant refusal for the accordion to open sans JS

## Additional information [optional]

* Taking a look at the changes in #1224 #1225 
*  `aria-expanded="false"` can be removed from any accordions we'd like open by default
* open the top accordion with  `aria-expanded="true"`

**Comment by joshbruce** [comment]
@rogeruiz - I believe the fix, based on the conversation in #1121, was to remove `aria-hidden="true"` from all accordions - have them all expanded by default - then let JS kick in and collapse.

Code accordions don't display without JS. All other accordions within the Standards are documentation blocks, which are set to `aria-hidden="false"` (expanded) anyway. So, it should be limited to the "Visual style" page.

I could either wait for feedback on #1244 and add the changes there. Or, make a new branch.

Slight concern on making a new branch since some of the line changes could cause a valid merge conflict beyond Screener failures.

Thoughts?

**Comment by joshbruce** [comment]
ps. The expand-collapse behavior still depends on JS modifying the `aria-hidden` attribute; so, leaving any accordions collapsed on load will result in not being able to expand.

**Comment by Drakenhart** [comment]
Have you considered pure CSS accordions? Is there a reason to rely so heavily on javascript for them? 

Pure CSS accordions have been made as far back as 2013 (although the ones I've seen are a bit "hacky" as they use a radio button or check box input that doesn't display visually). 

The accordion already uses a button element rather then a link.

Also is there a reason to use the Aria attribute as a hook? I'm curious as we have heard that often it is better to just visually hide elements (if needed), but display them to screen readers (ie don't hide them from the DOM). This was something we got feedback on in regards to pagination of data lists (don't hide them behind a next link - give them the whole page).

The only reason in *that* regard to limit on-screen output for screen readers is if it is too much of a data load for the server.

In this regards as an accordion, why use the aria attribute at all to hide the accordion content? Wouldn't that lead to more user frustration on a screen reader users part? Or do you have data that suggests otherwise? I'd love to know. :)

**Comment by joshbruce** [comment]
@Drakenhart - I agree - the checkbox (or radio button) is a hack - and potentially very confusing to users when their screen reader says there's a checkbox there without an associated form.

HTML 5.1 has, for the most part, an accordion element. Details + Summary - though support is the limiting factor there and would require a polyfill (JS solution anyway).

I did start working on a solution that used hover on the container if there was no JavaScript - if you had JS, then it would work on the click event. However, I believe the current is a solid one to address the JS and no-JS problem.

See #1192 for some discussion around the use of `aria-hidden`.

**Comment by Drakenhart** [comment]
@joshbruce Thank you for indulging me on those points. I appreciate it! It's sometimes hard to keep track of the various other conversations as well, so thank you for the link! :)

**Comment by jhfrench** [comment]
@joshbruce 

> I did start working on a solution that used hover on the container if there was no JavaScript 

Don't forget about mobile devices--they don't know about "hover" (well, not yet...and not reliably).

I think the safest approach is to load with all accordion children visible, and progressively-enhance to collapse only if the browser can support the relevant JS for expanding.

A CSS rule like this (assummes Modernizr syntax) would get you there:

```
HTML.js .usa-accordion-content.collapsed {
    //only hide accordion content if browser supports JS
    display: none;
}
```

**Comment by joshbruce** [comment]
@jhfrench - Agreed, and why I decided against submitting the PR. :) 

The accepted implementation is what you describe. #1224 

With the follow on #1247 to cover the accordions in the visual page.
